### PR TITLE
Prep for 4.1.5 release

### PR DESCRIPTION
**Pull request by MGatner**
Updates changelog and version references for 4.1.5.

### CODE REVIEW

File: user_guide_src/source/installation/upgrade_415.rst [hunk 1]
```diff
@@ -101,3 +101,41 @@ Content Changes
 
 All Changes
 -----------
```
Comment (kenjis) [inline user_guide_src/source/installation/upgrade_415.rst:103]: Remove this heading `All Changes` and the above `Content Changes`.
They are located below.

**Comment by kenjis** [comment]
It seems `user_guide_src/source/changelogs/index.rst` was updated when `Prep for 4.1.2`.
https://github.com/codeigniter4/CodeIgniter4/pull/4700/files#diff-381d2d09886543d18722e3aebb52071b41458d4659cc490ef63cb793d568b7e8

**Comment by kenjis** [comment]
I ran `github_changelog_generator`, but no `CHANGELOG.md` was generated.

```sh-session
$ docker run -it --rm -v "$(pwd)":/usr/local/src/your-app githubchangeloggenerator/github-changelog-generator --user codeigniter4 --project codeigniter4 --since-tag v4.1.4 --future-release v4.1.5 --token *****
Using these options:
:date_format=>"%Y-%m-%d"
:output=>"CHANGELOG.md"
:base=>"HISTORY.md"
:issues=>true
:add_issues_wo_labels=>true
:add_pr_wo_labels=>true
:pulls=>true
:filter_issues_by_milestone=>true
:issues_of_open_milestones=>true
:author=>true
:unreleased=>true
:unreleased_label=>"Unreleased"
:compare_link=>true
:exclude_labels=>["duplicate", "question", "invalid", "wontfix", "Duplicate", "Question", "Invalid", "Wontfix", "Meta: Exclude From Changelog"]
:summary_labels=>["Release summary", "release-summary", "Summary", "summary"]
:breaking_labels=>["backwards-incompatible", "Backwards incompatible", "breaking"]
:enhancement_labels=>["enhancement", "Enhancement", "Type: Enhancement"]
:bug_labels=>["bug", "Bug", "Type: Bug"]
:deprecated_labels=>["deprecated", "Deprecated", "Type: Deprecated"]
:removed_labels=>["removed", "Removed", "Type: Removed"]
:security_labels=>["security", "Security", "Type: Security"]
:configure_sections=>{}
:add_sections=>{}
:issue_line_labels=>[]
:max_issues=>nil
:simple_list=>false
:ssl_ca_file=>nil
:verbose=>true
:header=>"# Changelog"
:merge_prefix=>"**Merged pull requests:**"
:issue_prefix=>"**Closed issues:**"
:summary_prefix=>""
:breaking_prefix=>"**Breaking changes:**"
:enhancement_prefix=>"**Implemented enhancements:**"
:bug_prefix=>"**Fixed bugs:**"
:deprecated_prefix=>"**Deprecated:**"
:removed_prefix=>"**Removed:**"
:security_prefix=>"**Security fixes:**"
:http_cache=>true
:require=>[]
:config_file=>".github_changelog_generator"
:user=>"codeigniter4"
:project=>"codeigniter4"
:since_tag=>"v4.1.4"
:future_release=>"v4.1.5"
:token=>"hidden value"

WARNING: `Faraday::Connection#authorization` is deprecated; it will be removed in version 2.0.
While initializing your connection, use `#request(:authorization, ...)` instead.
See https://lostisland.github.io/faraday/middleware/authentication for more usage info.
Found 26 tags                                                          
Fetching tags dates: 26/26
Sorting tags...
Received issues: 5169
$
```

**Review by lonnieezell — APPROVED** [review]

File: user_guide_src/source/changelogs/v4.1.6.rst [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Version 4.1.6
```
Comment (paulbalandan) [inline user_guide_src/source/changelogs/v4.1.6.rst:1]: Include in toctree

File: user_guide_src/source/installation/upgrade_415.rst [hunk 3]
```diff
@@ -101,3 +101,41 @@ Content Changes
...
+* ``app/Config/Generators.php``
+* ``app/Config/Publisher.php``
+* ``app/Config/Security.php``
+* ``app/Views/welcome_message.php``
+
+All Changes
+-----------
+
+This is a list of all files in the project space that received changes;
+many will be simple comments or formatting that have no affect on the runtime:
```
Comment (paulbalandan) [inline user_guide_src/source/installation/upgrade_415.rst:132]:
```suggestion
many will be simple comments or formatting that have no effect on the runtime:
```

File: user_guide_src/source/changelogs/v4.1.6.rst (hunk 2 above)
Comment (MGatner) [reply user_guide_src/source/changelogs/v4.1.6.rst:1]: Thanks! Will do. This is what @kenjis was noting too.

**Review by paulbalandan — APPROVED** [review]

**Comment by MGatner** [comment]
Rebased with the `batch` PR from @kenjis

**Pull request merged**